lightkurve/correctors/designmatrix.py

```diff
@@ -212,10 +224,13 @@ def pca(self, nterms=6):
         # we find this to be too few, and that n_iter=10 is still fast but
         # produces more stable results.
         from fbpca import pca  # local import because not used elsewhere
-        new_values, _, _ = pca(self.values, nterms, n_iter=10)
-        return DesignMatrix(new_values, name=self.name)
+        if self._sparse:
+            new_values, _, _ = pca(self.values.toarray(), nterms, n_iter=10)
+        else:
+            new_values, _, _ = pca(self.values, nterms, n_iter=10)
+        return DesignMatrix(new_values, name=self.name, sparse=self._sparse)
```

Comment:
We could reduce code duplication above by adding a `self.get_values()` method to the class.  That method would take take care of returning either `self.values.toarray()` or `self.values` so other methods don't need to worry.

Reply:
Great idea.

Reply:
Ok so I'm thinking about this. Sometimes we want the sparse values, because we're going to do the linear algebra. But sometimes we want the numpy array of values, e.g. in the case of PCA. How do you want me to distinguish between these cases?

Reply:
e.g. there could be a 

`dm.X` for values OR sparse array, whatever is available 
and
`dm.values` for ALWAYS numpy arrays

Reply:
We should try to find a simple and consistent solution for this.  Should `DesignMatrix` perhaps always be sparse, with a method to populate a dense matrix as needed?